package/src/utilities/addDevToolbarPluginStubs/preact.ts

```diff
@@ -0,0 +1,22 @@
+import Component from "@@COMPONENT_SRC@@";
+import { h, render } from "preact";
+
+export default {
+	id: "@@ID@@",
+	name: "@@NAME@@",
+
+	// biome-ignore lint/style/noUnusedTemplateLiteral: Using backticks here because most likely copy+pasted svg paths will use double quotes
+	icon: `@@ICON@@`,
+	init: async (canvas) => {
+		const myWindow = document.createElement("astro-dev-toolbar-window");
+
+		canvas.appendChild(myWindow);
+
+		// biome-ignore lint/style/noUnusedTemplateLiteral: We want to be able to support multiline strings here
+		myWindow.insertAdjacentHTML("beforebegin", `<style>@@STYLE@@</style>`);
+
+		render(h(Component, {}, []), myWindow);
+
+		((canvas, window) => {})(canvas, myWindow); //@@CALLBACK@@
```

Comment:
tbh I'd prefer those 2 props to be passed as props to the component, would make things way cleaner. This way, there's no issue with callback serialization. We only need to provide a type for users like `DevtoolbarProps` whatever

Reply:
This also allows providing hooks or composables built on top of those props, eg. for easier communication with the vite dev server

Reply:
Done and done